etcdserver/server.go

```diff
@@ -1002,6 +1002,26 @@ func (s *EtcdServer) LeaderStats() []byte {
 
 func (s *EtcdServer) StoreStats() []byte { return s.store.JsonStats() }
 
+func (s *EtcdServer) checkMembershipOperationPermission(ctx context.Context) error {
```

Comment:
For example, we'll need changes in auth related functions of `applierV3Backend`? The changes won't be costly but will make this PR a little bit bloated.